mw-webapp/src/hooks/useLoad.ts

```diff
@@ -0,0 +1,76 @@
+import {useEffect, useState} from "react";
+
+/**
+ * Use Load hook Props
+ */
+interface useLoadProps<DataType, DependenciesType> {
+
+  /**
+   * Callback that is called to fetch data
+   */
+  loadData: () => Promise<DataType>;
+
+  /**
+   * Callback that is called to validate data
+   */
+  validateData?: (data: DataType) => boolean;
+
+  /**
+   * Callback that is called on fetch and validation success
+   */
+  onSuccess: (data: DataType) => void;
+
+  /**
+   * Callback this is called on fetch or validation error
+   */
+  onError?: (error: Error) => void;
+
+  /**
+   * Passed array of dependencies to re-fetch and re-validate data
+   */
+  dependencies: DependenciesType[];
+}
+
+/**
+ * Custom hook to fetch, validate, and manage the state of data
+ */
+export function useLoad<DataType, DependenciesType>({
+  loadData,
```

Comment:
Is it okey that hook useLoad not only fetch data but also valiadate it and other?

Reply:
We can rename it anytime, https://github.com/tritonJS826/masters-way/issues/300